server/api/db/sqldb/db.py

```diff
@@ -1293,10 +1294,8 @@ def _find_artifacts(
         most_recent: bool = False,
         attach_tags: bool = False,
         limit: int = None,
-    ) -> typing.Union[
-        list[tuple[ArtifactV2, str]],
-        list[ArtifactV2],
-    ]:
+        with_entities: list[Any] = None,
```

Comment:
what about using this instead of the `attach_tags` (putting the Tags in `with_entities` ?)

Reply:
that's not possible because the tags are joined in the inner query so we need to get the tag names from the subquey